examples/src/main/kotlin/io.zenoh/Config.kt

```diff
@@ -0,0 +1,78 @@
+//
+// Copyright (c) 2023 ZettaScale Technology
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License 2.0 which is available at
+// http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+//
+// Contributors:
+//   ZettaScale Zenoh Team, <zenoh@zettascale.tech>
+//
+
+package io.zenoh
+
+import io.zenoh.sample.Attachment
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.json.encodeToJsonElement
+import kotlin.io.path.Path
+
+@Serializable
+data class ConfigData(
+    @SerialName("connect") var connect: Connect? = null,
+    @SerialName("listen") var listen: Listen? = null,
+    @SerialName("mode") var mode: String? = null,
+    @SerialName("scouting") var scouting: Scouting? = null,
+)
+
+@Serializable
+data class Connect(
+    @SerialName("endpoints") var endpoints: List<String>
+)
+
+@Serializable
+data class Listen(
+    @SerialName("endpoints") var endpoints: List<String>
+)
+
+@Serializable
+data class Scouting(
+    @SerialName("multicast") var multicast: Multicast,
+)
+
+@Serializable
+data class Multicast(
+    @SerialName("enabled") var enabled: Boolean,
+)
+
+internal fun loadConfig(
+    emptyArgs: Boolean,
+    configFile: String?,
+    connectEndpoints: List<String>?,
+    listenEndpoints: List<String>?,
+    noMulticastScouting: Boolean,
+    mode: String?
+): Config {
+    val config = if (emptyArgs) {
+        Config.default()
+    } else {
+        configFile?.let { Config.from(Path(it)) } ?: run {
```

Comment:
This seems to only support full path. Is it possible to add support for relative path?

Reply:
Really? I test it with a relative path and it finds the file... Full path works as well.

Reply:
As discussed internally, the issue is that the path is taken relative to where the examples's `build.gradle.kts` file is located. I consider it to be an out of scope issue for this PR, however I have opened issue #68 to tackle this in the future.